test/libsolidity/ASTJSONTest.cpp

```diff
@@ -78,11 +103,25 @@ ASTJSONTest::ASTJSONTest(string const& _filename)
 	string_view baseName = _filename;
 	baseName.remove_suffix(4);
 
-	m_variants = {
-		TestVariant(baseName, CompilerStack::State::Parsed),
-		TestVariant(baseName, CompilerStack::State::AnalysisPerformed),
+	const std::vector<CompilerStack::State> variantCompileStates = {
+		CompilerStack::State::Parsed,
+		CompilerStack::State::AnalysisPerformed
 	};
 
+	for (const auto state: variantCompileStates)
+	{
+		auto variant = TestVariant(baseName, state);
+		if (boost::filesystem::exists(variant.astFilename()))
+		{
+			variant.expectation = readFileAsString(variant.astFilename());
+			boost::replace_all(variant.expectation, "\r\n", "\n");
+			m_variants.push_back(variant);
+		}
+	}
+
+	if (m_variants.empty())
+		BOOST_THROW_EXCEPTION(runtime_error("Missing file with expected result for: \"" + _filename + "\"."));
+
 	ifstream file(_filename);
```

Comment:
Is there any reason why all the stuff starting from here cannot be done by ``TestCaseReader`` instead? Not that it's strongly related to the issue at hand, but still...